apps/challenges/models.py

```diff
@@ -228,6 +228,7 @@ class StarChallenge(TimeStampedModel):
     """
     user = models.ForeignKey(User)
     challenge = models.ForeignKey(Challenge)
+    is_starred = models.BooleanField(default=False)
```

Comment:
Without this field when a user will unstar a challenge then we have to delete the entry from this model but now with this boolean field when a user will unstar it then we will toggle this field which will maintain our database schema indexes. Moreover, we can also get the analytics like how many people starred the challenge and how many are currently active.